### PR TITLE
[TUBEMQ-277] Codec Support, Encode and Decode implemented

### PR DESCRIPTION
**Pull request by lubanproj**
Codec Support, Encode and Decode implemented

### CODE REVIEW

**Comment by gosonzhang** [comment]
@lubanproj, there are some problems with the pr submitted

File: tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+func (d *pbSerialization) Unmarshal(data []byte, v interface{}) error {
+	if data == nil || len(data) == 0 {
+		return errors.New("unmarshal nil or empty bytes")
+	}
+
+	protoMsg := v.(proto.Message)
+	protoMsg.Reset()
+
+	if pu, ok := protoMsg.(proto.Unmarshaler); ok {
+		// 可以 unmarshal 自身，无需 buffer
```
Comment (liangyuanpeng) [inline tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go:88]: May be should not chinese. :)

File: tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+type pbSerialization struct{}
+
+func (d *pbSerialization) Marshal(v interface{}) ([]byte, error) {
+	if v == nil {
+		return nil, errors.New("marshal nil interface{}")
+	}
+	if pm, ok := v.(proto.Marshaler); ok {
+		// 可以 marshal 自身，无需 buffer
```
Comment (liangyuanpeng) [inline tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go:59]: May be should not chinese. :)

**Comment by liangyuanpeng** [comment]
We can code go client for TubeMQ together if you have any plan and agree. :)

File: tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
+package codec
```
Comment (liangyuanpeng) [inline tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go:1]: Need Licensed

File: tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go (hunk 1 above)
Comment (lubanproj) [reply tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go]: Thanks for the suggestion, it has been changed

File: tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go (hunk 3 above)
Comment (lubanproj) [reply tubemq-client-twins/tubemq-client-go/codec/codec/serialization.go]: Thanks for the suggestion, it has been changed

**Comment by gosonzhang** [comment]
Thanks @lubanproj and @liangyuanpeng

**Pull request merged**